unlock-app/src/components/interface/verification/invalidMembershipReason.ts

```diff
@@ -0,0 +1,37 @@
+import * as z from 'zod'
+import { isSignatureValidForAddress } from '~/utils/signatures'
+import { MembershipVerificationData } from '~/utils/verification'
+
+interface Options {
+  membershipData: any
+  rawData: string
+  data: z.infer<typeof MembershipVerificationData>
+  sig: string
+}
+
+export function invalidMembershipReason({
+  membershipData,
+  rawData,
+  data,
+  sig,
+}: Options) {
+  const { account, tokenId } = data
+  if (!isSignatureValidForAddress(sig, decodeURIComponent(rawData), account)) {
```

Comment:
It's important we make sure the changes that Kali worked on (to allow locksmith to be a signer) are used here.